opengrok-tools/src/main/python/opengrok_tools/utils/restful.py

```diff
@@ -18,31 +18,27 @@
 #
 
 #
-# Copyright (c) 2017, 2019, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2017, 2020, Oracle and/or its affiliates. All rights reserved.
 #
 
 import logging
 import json
+import requests
 
-from .webutil import put, post, delete
+from .webutil import get_proxies
 from .patterns import COMMAND_PROPERTY
 
 
 CONTENT_TYPE = 'Content-Type'
 APPLICATION_JSON = 'application/json'   # default
 
 
-def do_api_call(command, uri, verb, headers, data):
-    verbs = {
-        'PUT': put,
-        'POST': post,
-        'DELETE': delete
-    }
-    handler = verbs.get(verb)
-    if handler is not None:
-        logger = logging.getLogger(__name__)
-        return handler(logger, uri, headers=headers, data=data)
-    raise Exception('Unknown HTTP verb in command {}'.format(command))
+def do_api_call(uri, verb, headers=None, data=None):
```

Comment:
I believe you can pass the handler directly as `do_api_call('http:/...', requests.put, ... )` which could simplify the code around it

Reply:
Also I would switch the order to verb -> uri as that is the usual order

Reply:
Passing the handler directly would be arguably more error prone + would need a dependency in callers + is longer to type.

Reply:
changed the signature to use verb first.